src/Cms/Media.php

```diff
@@ -100,7 +100,7 @@ public static function thumb($model, string $hash, string $filename)
             $root = $kirby->root('media') . '/assets/' . $model . '/' . $hash;
         } else {
             // model files
-            $root = $model->mediaRoot() . '/' . $hash;
+            $root = dirname($model->mediaRoot());
```

Comment:
`mediaRoot` already contains both the filename and the hash.

**Current Path**
    /media/pages/photography/animals/22b8cf85b7-1600414901/peter-fox-38x38.jpg/22b8cf85b7-1600414901/.jobs/peter-fox-38x38.jpg.json

**Improved Path**
    /media/pages/photography/animals/22b8cf85b7-1600414901/.jobs/peter-fox-38x38.jpg.json

Reply:
Are you sure about this? I checked the mediaRoot method for the Page class again for example and it does not include the hash. So it would be wrong for page files for example.

Reply:
Hash is included, check out please:

https://github.com/getkirby/kirby/blob/3.4.3/src/Cms/File.php#L360-L363

Reply:
Job file like following, is that normal?

> /tests/Cms/Media/fixtures/MediaTest/media/site/c87dff63b0-1600766405/test.jpg/c87dff63b0-1600766405/.jobs/test.jpg.json

Reply:
@bastianallgeier What do you think about that?

Reply:
I think you might be right. The weird thing is that I wasn't able to find any place where we use Media::thumb with a File model as first parameter. Did you find a place for that?

Reply:
I could not find it either. It doesn't need hash in the path of other objects but `File` do. I guess that's why it contains hashes.

How about if we write it like this? Or if you have a better idea?

````php
if (is_string($model) === true) {
    // assets
    $root = $kirby->root('media') . '/assets/' . $model . '/' . $hash;
} elseif (is_a($model, '\Kirby\Cms\File')) {
    // model files for file model that already included hash
    $root = dirname($model->mediaRoot());
} else {
    // model files
    $root = $model->mediaRoot() . '/' . $hash;
}
````

Reply:
I like that a lot! Could you make the change?

Reply:
Great! Yes, i'll.